Autocoders/Python/src/fprime_ac/parsers/XmlSerializeParser.py

```diff
@@ -169,23 +169,48 @@ def __init__(self, xml_file=None):
                         sys.exit(-1)
                     n = member.attrib["name"]
                     t = member.attrib["type"]
+                    if "array_size" in list(member.attrib.keys()):
+                        if t == "ENUM":
+                            PRINT.info(
+                                "%s: Member %s: arrays of enums not supported yet!"
+                                % (xml_file, n)
+                            )
+                            sys.exit(-1)
+                        array_size = member.attrib["array_size"]
+                        if not array_size.isdigit():
+                            PRINT.info(
+                                "{}: Member {}: array_size must be a number".format(
+                                    xml_file, n
+                                )
+                            )
+                            sys.exit(-1)
+                    else:
+                        array_size = None
+
                     if "size" in list(member.attrib.keys()):
                         if t == "ENUM":
                             PRINT.info(
                                 "%s: Member %s: arrays of enums not supported yet!"
                                 % (xml_file, n)
                             )
                             sys.exit(-1)
-                        s = member.attrib["size"]
-                        if not s.isdigit():
+                        size = member.attrib["size"]
+                        if not size.isdigit():
                             PRINT.info(
-                                "{}: Member {}: size must be a number".format(
+                                "{}: Member {}: array_size must be a number".format(
                                     xml_file, n
                                 )
                             )
                             sys.exit(-1)
+                        if t != "string":
```

Comment:
SHouldn't this block exit with an  as on line 204, or otherwise produce an error?

Reply:
Yes, good catch. I'll make that fix.